core/src/main/resources/core-migrations/local/V3__Logging.sql

```diff
@@ -1,7 +1,51 @@
 create table if not exists logs
 (
-    id        int auto_increment primary key,
+    id        varchar(255)                       primary key,
     Level     varchar(255)                          not null,
     Message   text                                  not null,
-    Timestamp timestamp default current_timestamp() not null
-);
\ No newline at end of file
+    Time      bigint                                not null
+);
+
+create table if not exists uuiditems
+(
+    UUID        varchar(255)    PRIMARY KEY,
+    Namespace   varchar(255)    NOT NULL,
+    Keyname     varchar(255)    NOT NULL
+);
+
+create table if not exists uuidlogmeta
+(
+    id          varchar(255)    PRIMARY KEY,
+    LogUUID     varchar(255)    not null,
+    ItemUUID    varchar(255)    not null,
+    Type        varchar(255)    not null,
+    UUID        varchar(255)    default null,
+    UUIDtype    varchar(255)    default null,
+
+
+    CONSTRAINT uuidlogmeta_id_fk
+        FOREIGN KEY (LogUUID) REFERENCES logs (id),
+    CONSTRAINT uuidlogmeta_uuid_fk
+        FOREIGN KEY (ItemUUID) REFERENCES uuiditems (UUID)
+);
+
+DROP PROCEDURE IF EXISTS GetUuidLogsByUuid;
+CREATE PROCEDURE GetUuidLogsByUuid(UniqueID varchar(255), amount int)
+BEGIN
+    SELECT DISTINCT Time, Message FROM logs, uuidlogmeta
+        WHERE ItemUUID = UniqueID
+        AND LogUUID = logs.id
+        ORDER BY Time DESC
+        LIMIT amount;
+END;
+
+DROP PROCEDURE IF EXISTS GetUuidLogsByPlayer;
+CREATE PROCEDURE GetUuidLogsByPlayer(PlayerUuid varchar(255), amount int)
+BEGIN
+    SELECT DISTINCT Time, Message FROM logs, uuidlogmeta
+        WHERE UUID = PlayerUuid
+        AND UUIDtype = 'PLAYER'
+        AND LogUUID = logs.id
+        ORDER BY Time DESC
+        LIMIT amount;
+END;
```

Comment:
A little strange, can you talk me through what kind of data you expect these to return, and why you need to check the LogUUID?

Reply:
This returns the Time, Message of the UUID/Player (depending on function) for all logs with the relevant UUID/Player, ordered by most recent first.

We need to check the LogUUID so that the log being retrieved is the log the meta data is about.